includes/admin/upgrades/classes/class-file-download-log-migration.php

```diff
@@ -0,0 +1,286 @@
+			foreach ( $step_items as $log_id ) {
+
+				$sanitized_log_id = absint( $log_id );
+
+				if ( $sanitized_log_id !== $log_id ) {
+					edd_debug_log( "Log ID mismatch, skipping log ID {$log_id}" );
+					continue;
+				}
+
+				$payment_id = $wpdb->get_var( "SELECT meta_value FROM {$wpdb->postmeta} WHERE post_id = {$sanitized_log_id} AND meta_key = '_edd_log_payment_id'" );
+				if ( ! empty( $payment_id ) ) {
+					$payment     = edd_get_payment( $payment_id );
+					$customer_id = $payment->customer_id;
+
+					if ( $customer_id < 0 ) {
+						$customer_id = 0;
+					}
+
+					edd_debug_log( "Updating log ID {$log_id} with customer ID {$customer_id} and removing user info" );
+
+					update_post_meta( $log_id, '_edd_log_customer_id', $customer_id );
+					delete_post_meta( $log_id, '_edd_log_user_info' );
+
+				}
+
+			}
+
+			return true;
+
+		}
+
+		return false;
+
+	}
+
+	/**
+	 * Return the calculated completion percentage
+	 *
+	 * @since 2.9.2
+	 * @return int
+	 */
+	public function get_percentage_complete() {
+
+		$items = $this->get_stored_data( 'edd_file_download_log_ids', false );
+		$total = count( $items );
+
+		$percentage = 100;
+
+		if( $total > 0 ) {
+			$percentage = ( ( $this->per_step * $this->step ) / $total ) * 100;
+		}
+
+		if( $percentage > 100 ) {
+			$percentage = 100;
+		}
+
+		return $percentage;
+	}
+
+	/**
+	 * Set the properties specific to the payments export
+	 *
+	 * @since 2.9.2
+	 * @param array $request The Form Data passed into the batch processing
+	 */
+	public function set_properties( $request ) {}
+
+	/**
+	 * Process a step
+	 *
+	 * @since 2.9.2
+	 * @return bool
+	 */
+	public function process_step() {
+
+		if ( ! $this->can_export() ) {
+			wp_die(
+				__( 'You do not have permission to run this upgrade.', 'easy-digital-downloads' ),
+				__( 'Error', 'easy-digital-downloads' ),
+				array( 'response' => 403 ) );
+		}
+
+		$had_data = $this->get_data();
+
+		if( $had_data ) {
+			$this->done = false;
+			return true;
+		} else {
+			$this->delete_data( 'edd_file_download_log_ids' );
+
+			$this->done    = true;
+			$this->message = __( 'File download logs updated successfully.', 'easy-digital-downloads' );
+			edd_set_upgrade_complete( 'update_file_download_log_data' );
+			return false;
+		}
+	}
+
+	public function headers() {
+		ignore_user_abort( true );
+
+		if ( ! edd_is_func_disabled( 'set_time_limit' ) ) {
+			set_time_limit( 0 );
+		}
+	}
+
+	/**
+	 * Perform the export
+	 *
+	 * @access public
+	 * @since 2.9.2
+	 * @return void
+	 */
+	public function export() {
+
+		// Set headers
+		$this->headers();
+
+		edd_die();
+	}
+
+	public function pre_fetch() {
+		global $edd_logs;
+
+		if ( $this->step == 1 ) {
+			$this->delete_data( 'edd_file_download_log_ids' );
+		}
+
+		// Get any file download logs that have an entry for the user info
+		$file_download_log_ids = get_option( 'edd_file_download_log_ids', false );
+		if ( false === $file_download_log_ids ) {
+
+			$log_query = array(
```

Comment:
I doubt this will be successful on really big sites that have hundreds of thousands of file download log entries. Can we move the query logic into the step processor?

Reply:
@pippinsplugins Possibly, I'll have to play around with it to make sure the pagination still works.

Reply:
If unable to make pagination work, let's at minimum test this on sites with x00,000 download logs.